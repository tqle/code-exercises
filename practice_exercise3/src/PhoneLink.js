import React from "react";
import { validatePhone } from "./utils/linkUtil";
/**
 * Phone Link extends link tag with
 * custom tel: href to indicate link is telephone number.
 */
class PhoneLink extends React.Component {
  render() {
    // Destructuring objects from props.
    let { phone } = this.props;
    // Check for empty, undefined, null cases
    let validPhone = validatePhone(phone);
    // FUTURE: Write a regex or pattern match if phone is valid phone format
    let title = validPhone ? "Telephone:" + validPhone : "";
    let linkReference = validPhone ? "tel:" + validPhone : "#";

    return (
      <a title={title} href={linkReference}>
        {validPhone}
      </a>
    );
  }
}
export default PhoneLink;
