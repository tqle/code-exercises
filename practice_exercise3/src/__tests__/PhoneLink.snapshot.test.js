import React from "react";
import { create } from "react-test-renderer";
import PhoneLink from "../PhoneLink";

/**
 * Test Phone Link snapshot
 */
test("<PhoneLink />renders correctly", () => {
  const phone = "493-170-9623 x156";
  const test = create(<PhoneLink phone={phone} />);
  expect(test.toJSON()).toMatchSnapshot();
});

test("<PhoneLink /> renders as an anchor when no phone number reference is set", () => {
  const phone = "493-170-9623 x156";
  const test = create(<PhoneLink>{phone}</PhoneLink>).toJSON();
  expect(test).toMatchSnapshot();
});
