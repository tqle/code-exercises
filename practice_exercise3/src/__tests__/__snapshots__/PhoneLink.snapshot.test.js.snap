// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneLink /> renders as an anchor when no phone number reference is set 1`] = `
<a
  href="#"
  title=""
>
  
</a>
`;

exports[`<PhoneLink />renders correctly 1`] = `
<a
  href="tel:493-170-9623 x156"
  title="Telephone:493-170-9623 x156"
>
  493-170-9623 x156
</a>
`;
